### ISSUE TITLE
Additional Metrics for System.Net.Http

### ISSUE THREAD

**Issue by antonfirsov** [dotnet/runtime#88384]
Opening a separate issue for the HTTP part of #79371, also including requests from #79459. We should eventually implement the Metrics counterpart of all [System.Net.Http EventCounters](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/available-counters#systemnethttp-counters).

#### The following counters already have their Metrics counterparts specified and tracked by #84978
- `requests-failed`, `current-requests`
- `http11-connections-current-total`, `http20-connections-current-total`, `http30-connections-current-total`
- `http11-requests-queue-duration`, `http20-requests-queue-duration`, `http30-requests-queue-duration`

#### Event counters not discussed in #84978
- `requests-started`
- `requests-started-rate`
- `requests-failed-rate`

#### Additional requests from #79459, not listed above
- number of idle connections
- average idle connection time
- number of connections evicted from pool due to idle timeout
- number of connection timeouts
- average time to connect
- queue length
- number of http 2 keep alive pings which didn't receive a response
- average response time of http 2 keep alive pings

#### Regarding tags, there is a wish in #79459:

> Ideally these metrics would be available per FQDN:Port combination, broken down by type (i.e. HTTP1.1/HTTP2) and there would be some way to track which http client each pool belongs to.

**Comment by antonfirsov** [comment]
@davidfowl all connection instruments from #84978 have been added now. I mistakenly though that `http(11|20|30)-requests-queue-duration` is also tracked there which is not the case, and it fits the definition of *the connection pooling data that exists inside of SocketHttpHandler's guts* from https://github.com/dotnet/runtime/issues/48885#issuecomment-1631589033. Do you find this critical for 8.0?

**Comment by davidfowl** [comment]
Yes, that's *very* important to add. One of the most important counters when looking at slownes of requests.

**Comment by MihaZupan** [comment]
`requests-queue-duration` has been added in #88981
